CMakeLists.txt

```diff
@@ -1891,6 +1891,21 @@ if(CONFIG_BUILD_OUTPUT_INFO_HEADER)
     )
 endif()
 
+if (CONFIG_SYSCALL_STACK_USAGE)
+  list(APPEND
+    post_build_commands
+    COMMAND ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/stack_usage/stack_usage.py
+    --elf-file=${KERNEL_ELF_NAME}
+    --output-file=${PROJECT_BINARY_DIR}/syscall_stack_usage.txt
+    --build-dir=${PROJECT_BINARY_DIR}
+    --input-flags=${CONFIG_SYSCALL_STACK_USAGE_OPTIONS}
+    )
```

Comment:
the output from `stack_usage` is not going to be used by the build itselfn or is it providing something directly related to the build, such as a conversion of the elf files into bin / hex.

Also the ability to generate a syscall stack usage should not require a rebuilt just to be able to get this information (changing a Kconfig setting will cause a rebuild).

Please make this a custom target, similar to how `ram/rom_report`, `puncover`, `pahole` works:
https://github.com/zephyrproject-rtos/zephyr/blob/44d101acabb5f3c0916ba9c193361c32e16c015c/cmake/reports/CMakeLists.txt#L14-L16